### PR TITLE
[Impeller] Cache entire render target and not just allocations.

### PR DESCRIPTION
**Pull request by jonahwilliams**
This may be required to fix https://github.com/flutter/flutter/issues/144116 , and in general the safety of the render pass caches.

Because the RenderPass / Framebuffer objects refer to specific attachments, the caches are placed on the resolve texture, which is "real". The stencil/depth and or MSAA textures may be different but that didn't seem to matter on many Android devices. It doesn't seem like its actually valid though, whether or not there is a validation error to catch it.

I can confirm this fixes https://github.com/flutter/flutter/issues/144116  locally.

This change moves the cache one level up to cover the properties of the RenderTarget we care about, and ensures that we always use the same attachment combinations.

Fixes https://github.com/flutter/flutter/issues/141750
Maybe https://github.com/flutter/flutter/issues/144255 ?

### CODE REVIEW

File: impeller/renderer/render_target.h [hunk 1]
```diff
@@ -18,32 +18,21 @@ namespace impeller {
...
-  virtual void Start();
-
-  /// @brief Mark the end of a frame workload.
-  ///
-  ///        This may be used to deallocate any unused textures.
-  virtual void End();
-
- private:
-  std::shared_ptr<Allocator> allocator_;
+struct RenderTargetConfig {
```
Comment (jonahwilliams) [inline impeller/renderer/render_target.h:21]: This is a subset of the render target properties that we care about for caching (at the entity level).

Comment (jonahwilliams) [reply impeller/renderer/render_target.h:21]: Its defined here for ... reasons, but I could move this up to the entity cache so its more obvious

**Review by dnfield — APPROVED** [review]
Was there a DISABLED test you can re-enable now?

File: impeller/renderer/backend/vulkan/render_pass_cache_unittests.cc [hunk 2]
```diff
@@ -11,7 +11,41 @@
 namespace impeller {
 namespace testing {
 
-//
+using RendererTest = PlaygroundTest;
```
Comment (jonahwilliams) [inline impeller/renderer/backend/vulkan/render_pass_cache_unittests.cc:14]: @dnfield  See here. For some reasonable using DISABLE_ caused a problem with the test harness so I just commented it out.

**Review by bdero — APPROVED** [review]
LGTM

**Pull request merged**